draft-bft-rats-kat.md

```diff
@@ -55,36 +55,148 @@ This document defines an evidence format for key attestation.
+The KAT is signed by the KAK. The key attestation service, which is part of the platform 
+root of trust (RoT), conceptually acts as a local certification authority since the KAT 
+behaves like a certificate.
+
+- Combined Attestation Bundle (CAB): A structure used to bundle a KAT and a PAT together
+for transport in the usage protocol. If the KAT already includes a PAT, in form of a
+nested token, then it already corresponds to a CAB.
+
+- Presenter: Party that proves possession of a private key to a recipient of a KAT.
+
+- Recipient: Party that receives the KAT containing the proof-of-possession key information 
+from the presenter.
+
+- Key Attestation Service (KAS): The issuer that creates the KAT and bundles a KAT together
+with a PAT in a CAB.
+
+The reader is assumed to be familiar with the vocabulary and concepts defined in {{-rats-arch}}.
+
+# Architecture
+
+In an architecture for key attestation there are three or more parties communicating with
+each other. From a key attestation point of view, there are the Key Attestation Service,
+the Presenter and the Recipient. Additional parties are added when considering attestation-
+related entities.
+
+~~~ 
+ +----------------------------------+
+ | Attester                         |
+ |                                  |
+ | +-------------+  +-------------+ |
+ | | Key         |  | Platform    | |
+ | | Attestation |  | Attestation | |
+ | | Service     |  | Service     | |
+ | +-------------+  +-------------+ |
+ +----------------------------------+
+       ^
+       |
+       v
+ +-----------------+                 +-----------------+
+ |                 |                 |                 |
+ |   Presenter     | Usage Protocol  |   Recipient     |
+ |                 |---------------->|                 |
+ +-----------------+                 +-----------------+
+~~~
+{: #fig-arch title="Architecture"}
 
-The examples in {{examples}} use CBOR diagnostic notation defined in {{Section
-8 of -cbor}} and {{Appendix G of -cddl}}.
+The Presenter generates the IK, which contains a public key (pkT) and a private key (skT), for example
+using the following API call:
 
-The reader is assumed to be familiar with the vocabulary and concepts
-defined in {{-rats-arch}}.
+~~~~
+key_id = GenerateKeyPair(alg_id)
+~~~~
 
-# Key Attesatation Token
+The private key is created and stored such that it is only accessible to the 
+KAS rather than to the Presenter.
 
-## Proof-of-Possession
+Next, the KAS needs to trigger the creation of the Platform
+Attestation Token (PAT) by the Platform Attestation Service component. The PAT needs 
+to be linked to the Key Attestation Token (KAT). The Key Attestation Token (KAT) includes
+the public key of the IK (pkT) and is then signed with the Key Attestation Key (KAK).
+
+To ensure freshness of the PAT and the KAT a nonce is used, as suggested by the RATS
+architecture. Here is the symbolic API call to request a KAT and a PAT, which 
```

Comment:
Regarding the freshness - do we want to convey the option of "PAT generated at initialisation and then cached", where only the KAT includes the freshness? Also, I don't think it's specified that the RP is the one providing the nonce, but perhaps that can be assumed

Reply:
@ionut-arm, do you think we could convert this into an issue to be handled subsequently and merge as-is?

Reply:
Sure!